pkg/builder3/openapi.go

```diff
@@ -0,0 +1,340 @@
+package builder3
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	restful "github.com/emicklei/go-restful"
+	"k8s.io/kube-openapi/pkg/common"
+	"k8s.io/kube-openapi/pkg/spec3"
+	"k8s.io/kube-openapi/pkg/util"
+	"k8s.io/kube-openapi/pkg/validation/spec"
+	"strings"
+)
+
+const (
+	OpenAPIVersion = "3.0"
+)
+
+type openAPI struct {
+	config      *common.Config
+	spec        *spec3.OpenAPI
+	definitions map[string]common.OpenAPIDefinition
```

Comment:
There's a GetDefinitions function interface that we still rely on (https://github.com/kubernetes/kube-openapi/blob/master/pkg/common/common.go#L87), and I thought it might be good to keep consistency. This struct technically holds "OpenAPIDefinition" objects that have not been converted to their respective components yet so I think the naming is still accurate?